snapshot/snapshot.go

```diff
@@ -282,78 +282,81 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 		}
 	}
 
-	if target, ok := base.Labels[targetSnapshotLabel]; ok {
-
-		// NOTE: If passed labels include a target of the remote snapshot, `Prepare`
-		//       must log whether this method succeeded to prepare that remote snapshot
-		//       or not, using the key `remoteSnapshotLogKey` defined in the above. This
-		//       log is used by tests in this project.
-		lCtx := log.WithLogger(ctx, log.G(ctx).WithField("key", key).WithField("parent", parent))
-
-		localOverride := false
-		if o.minLayerSize > 0 {
-			if strVal, ok := base.Labels[source.TargetSizeLabel]; ok {
-				if intVal, err := strconv.ParseInt(strVal, 10, 64); err == nil {
-					if intVal < o.minLayerSize {
-						localOverride = true
-						log.G(lCtx).Info("Layer size less than runtime min_layer_size, skipping remote snapshot preparation")
-					}
-				} else {
-					log.G(lCtx).WithError(err).Error("Config min_layer_size cannot be converted to int")
-				}
-			}
-		}
-
-		if !localOverride {
-			if err := o.prepareRemoteSnapshot(lCtx, key, base.Labels); err != nil {
-				log.G(lCtx).WithField(remoteSnapshotLogKey, prepareFailed).
-					WithError(err).Warn("failed to prepare remote snapshot")
-				if !errors.Is(err, ErrNoZtoc) {
-					commonmetrics.IncOperationCount(commonmetrics.FuseMountFailureCount, digest.Digest(""))
-				}
-			} else {
-				base.Labels[remoteLabel] = remoteLabelVal // Mark this snapshot as remote
-				err := o.commit(ctx, true, target, key, append(opts, snapshots.WithLabels(base.Labels))...)
-				if err == nil || errdefs.IsAlreadyExists(err) {
-					// count also AlreadyExists as "success"
-					log.G(lCtx).WithField(remoteSnapshotLogKey, prepareSucceeded).Info("Remote snapshot successfully prepared.")
-					return nil, fmt.Errorf("target snapshot %q: %w", target, errdefs.ErrAlreadyExists)
-				}
-				log.G(lCtx).WithField(remoteSnapshotLogKey, prepareFailed).
-					WithError(err).Warn("failed to internally commit remote snapshot")
-				// Don't fallback here (= prohibit to use this key again) because the FileSystem
-				// possible has done some work on this "upper" directory.
-				return nil, err
-			}
-		}
-
-		mounts, err := o.mounts(ctx, s, parent)
```

Comment:
In cases where the current impl calls `o.mounts` twice, I assume it's not on purpose (e.g., between here and final `return o.mounts`, we change `s` or `parent` or something `s`/`parent` rely on so `o.mounts` has to be called again at the end.